### PR TITLE
cmake: simplify highlight queries

### PR DESCRIPTION
**Pull request by uyha**
Currently, the highlight queries try to match the commands with their keywords according to the CMake document. This approach performs poorly with the current state of nvim's tree-sitter. So I simplify it for now to just assume that unquoted arguments are keywords for function calls. Since this is the convention for CMake so I hope that it works well. This should also improve the startup time when opening CMake files.

### CODE REVIEW

**Comment by amaanq** [comment]
I don't get the point of this, are you experiencing major editor slowdown/lag with the current queries? Do you have a reproducible file to test on?

**Comment by uyha** [comment]
well, first is this [issue](https://github.com/nvim-treesitter/nvim-treesitter/issues/1956) (pun intended) and then the startup time when opening a CMake is noticable also as indicated [here](https://github.com/uyha/tree-sitter-cmake/issues/11).

**Comment by uyha** [comment]
> I still think this removes a lot of functionality. Some of the bigger matches/any-ofs could be removed, sure, but try incrementally testing with smaller ones..like:
> 
> (normal_command
>   (identifier) @_function
>   (#match? @_function "\\c^unset$")
>   . (argument)
>   (argument) @storageclass
>   (#any-of? @storageclass "CACHE" "PARENT_SCOPE")
> )
Oh, I totally forgot that I have those queries there also, thanks for catching that.

> And for any [A-Z]* keywords (all caps), what do you think about default highlighting it as @constant? (Before the matches due to query precedence)
I'm not sure i understand this correctly, do you mean to move the query to the beginning so it can be overridden by following queries?

**Comment by amaanq** [comment]
> I'm not sure i understand this correctly, do you mean to move the query to the beginning so it can be overridden by following queries?

Yes, it'd be worthwhile have a general [A-Z] constant match at the beginning (or at least before all the in-depth matches)

For example, there seems to be this:
```scheme
(normal_command
  (identifier)
  (argument (unquoted_argument)) @constant
  (#match? @constant "^[A-Z][A-Z_]+$")
)
```
Which is pretty much what I was talking about, except for the actual identifier instead of argument as well.

**Comment by uyha** [comment]
> Which is pretty much what I was talking about, except for the actual identifier instead of argument as well.

Oh no, `argument` is what needs to be there, not `identifier`. For example, given `message(STATUS Hello)`, the `identifier` in this case refers to `message`, while `argument` will refer to `STATUS` and `Hello`. What should be highlighted as a `@constant` is `STATUS` not `message`.

**Comment by amaanq** [comment]
The first issue is definitely weird, 

> 

Oh, maybe I misinterpreted that query wrong.  Say we have a function/rule, any CAPS identifiers inside that function/rule (or at the top level) should be `@constant`

**Comment by uyha** [comment]
Maybe the name is confusing but `identifier` does not bind to the arguments in the function call. It only binds to the name of the function, so there is no `identifier` inside that function.

**Comment by uyha** [comment]
but they are in the grammar already

File: queries/cmake/highlights.scm [hunk 1]
```diff
@@ -1,3 +1,9 @@
+(normal_command
+  (identifier)
+  (argument (unquoted_argument)) @constant
+  (#match? @constant "^[A-Z][A-Z_]+$")
```
Comment (theHamsta) [inline queries/cmake/highlights.scm:4]: Does this work also with lua-match?

Comment (uyha) [reply queries/cmake/highlights.scm:4]: I am not sure what you mean

**Review by theHamsta — APPROVED** [review]

File: queries/cmake/highlights.scm (hunk 1 above)
Comment (theHamsta) [reply queries/cmake/highlights.scm:4]: `lua-match?` uses the Lua implementation fro regular expressions which is more efficient than vim's `match?`. `match?` supportsore features than Lua patterns

Sorry! I forgot about this comment. The PR looks great apart from that minor considation

Comment (uyha) [reply queries/cmake/highlights.scm:4]: oh, I have no idea about `lua-match?`, I will try it out.

**Comment by ObserverOfTime** [comment]
> but they are in the grammar already

I meant `NOT`, `AND`, `OR`, etc.

**Pull request merged**